### PR TITLE
12.0 signatory inhabitant

### PR DESCRIPTION
**Pull request by mclaeysb**
- Adds signatories and inhabitants to leases
- Facilitates lease search by tenant, signatory and inhabitant
- Adds checkmark on leases when they are shared within cluster (anticipating rework of cluster logic)

### CODE REVIEW

File: housing_cooperative_base/demo/demo.xml [hunk 1]
```diff
@@ -121,7 +121,7 @@
         </record>
 
         <record id="demo_lease_1" model="hc.lease">
-            <field name="tenant_id" ref="res_partner_cooperator_1_demo"/>
```
Comment (robinkeunen) [inline housing_cooperative_base/demo/demo.xml]: Il faut quand même laisser un tenant, non?

File: housing_cooperative_base/models/lease.py [hunk 2]
```diff
@@ -140,6 +150,28 @@ def _compute_state(self):
             else:
                 lease.state = "draft"
 
+    @api.multi
+    @api.depends("lease_line_ids")
```
Comment (robinkeunen) [inline housing_cooperative_base/models/lease.py:169]: I think you need to depend on `lease_line_ids.premise_id`

File: housing_cooperative_base/models/lease.py [hunk 3]
```diff
@@ -140,6 +150,28 @@ def _compute_state(self):
...
+                )
+            )
+
+    @api.onchange("signatory_ids")
+    def onchange_signatory_ids(self):
+        self.inhabitant_ids |= self.signatory_ids
+
+    @api.onchange("inhabitant_ids")
+    def onchange_inhabitant_ids(self):
+        self.inhabitant_ids |= self.signatory_ids
```
Comment (robinkeunen) [inline housing_cooperative_base/models/lease.py]: There are probably conflicts or loops here.

**Review by robinkeunen — COMMENTED** [review]
Preliminary comments, let's talk about the rest on Monday :-)

File: housing_cooperative_base/demo/demo.xml (hunk 1 above)
Comment (mclaeysb) [reply housing_cooperative_base/demo/demo.xml]: En effet, confirmé avec Michel!

File: housing_cooperative_base/models/lease.py (hunk 3 above)
Comment (mclaeysb) [reply housing_cooperative_base/models/lease.py]: Cleaner in upcoming commits!

File: housing_cooperative_base/models/lease.py (hunk 2 above)
Comment (mclaeysb) [reply housing_cooperative_base/models/lease.py:169]: You found a bug! 👍 
The field is computed correctly when a housing is added to a lease, but the depends does not detect a change in the value of the 'is_arcade' field in the housing later in, not even if you extend it as you propose. I might rewrite this part (later) to put this field in the premise, which is the only way I currently see to point to the actual field in the depends... Ideas welcome.

Comment (mclaeysb) [reply housing_cooperative_base/models/lease.py:169]: Just tried that and it didn't solve it. It's hard to make it check for a mapped field... Ideas?

Comment (robinkeunen) [reply housing_cooperative_base/models/lease.py:169]: Let's have a chat about it. :-)

**Pull request merged**